lib/socket.c

```diff
@@ -1050,12 +1078,17 @@ smb2_connect_async(struct smb2_context *smb2, const char *server,
         if (err != 0) {
 #endif
                 free(addr);
-#ifdef _WINDOWS
+#if defined(_WINDOWS) || defined(_XBOX)
                 if (err == WSANOTINITIALISED)
                 {
-                        smb2_set_error(smb2, "Winsock was not initialized. "
+#ifdef _XBOX
+  					smb2_set_error(smb2, "Winsockx was not initialized. "
+                                "Please call WSAStartup().");                      
+#else
+					smb2_set_error(smb2, "Winsock was not initialized. "
```

Comment:
Lets not have a big block of code and ifdefs just to change an error string that the user will likely not see anyway.
So get rid of this change completely.

We should have as little #if defined(platform)   as possible in the general *.c files as possible